src/coreclr/jit/importercalls.cpp

```diff
@@ -4169,6 +4154,328 @@ GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic        intrinsic,
+
+            if (op1->IsIntegralConst())
+            {
+                if (varTypeIsLong(retType))
+                {
+                    uint64_t cns = static_cast<uint64_t>(op1->AsIntConCommon()->LngValue());
+
+                    if (varTypeIsSigned(baseType) && (static_cast<int64_t>(cns) < 0))
+                    {
+                        // Throwing case, fallback to software impl
+                        return nullptr;
+                    }
+
+                    return gtNewIconNode(BitOperations::Log2(cns), retType);
+                }
+                else
+                {
+                    uint32_t cns = static_cast<uint32_t>(op1->AsIntConCommon()->IconValue());
+
+                    if (varTypeIsSigned(baseType) && (static_cast<int64_t>(cns) < 0))
+                    {
+                        // Throwing case, fallback to software impl
+                        return nullptr;
+                    }
+
+                    return gtNewIconNode(BitOperations::Log2(cns), retType);
+                }
+            }
+
+            if (varTypeIsSigned(baseType))
+            {
+                // Signed needs to throw for negative inputs, so fallback to software impl
+                return nullptr;
+            }
+
+#if defined(FEATURE_HW_INTRINSICS)
+            GenTree* lzcnt = impPrimitiveNamedIntrinsic(NI_PRIMITIVE_LeadingZeroCount, clsHnd, method, sig);
+
+            if (lzcnt != nullptr)
+            {
+                GenTree* icon = gtNewIconNode(varTypeIsLong(retType) ? 63 : 31, retType);
+                return gtNewOperNode(GT_XOR, retType, lzcnt, icon);
+            }
+#endif // FEATURE_HW_INTRINSICS
+
+            return nullptr;
+        }
+
+        case NI_PRIMITIVE_PopCount:
+        {
+            assert(sig->numArgs == 1);
+
+            var_types baseType = JitType2PreciseVarType(sig->retType);
+            assert(!varTypeIsSmall(baseType));
+
+            GenTree* op1 = impStackTop().val;
+
+            if (op1->IsIntegralConst())
+            {
+                if (varTypeIsLong(retType))
+                {
+                    uint64_t cns = static_cast<uint64_t>(op1->AsIntConCommon()->LngValue());
+                    return gtNewIconNode(BitOperations::PopCount(cns), retType);
+                }
+                else
+                {
+                    uint32_t cns = static_cast<uint32_t>(op1->AsIntConCommon()->IconValue());
+                    return gtNewIconNode(BitOperations::PopCount(cns), retType);
+                }
+            }
+
+#if defined(FEATURE_HW_INTRINSICS)
+#if defined(TARGET_XARCH)
+            if (compOpportunisticallyDependsOn(InstructionSet_POPCNT))
+            {
+                // Pop the value from the stack
+                impPopStack();
+
+                hwintrinsic = varTypeIsLong(retType) ? NI_POPCNT_X64_PopCount : NI_POPCNT_PopCount;
+                return gtNewScalarHWIntrinsicNode(retType, op1, hwintrinsic);
+            }
```

Comment:
Since there isn't an "always available" instruction for x86/x64, we should probably import this as `GenTreeIntrinsic` much as happens for various Math APIs like `Sin`, `Cos`, and other APIs.

Doing so would allow us to still perform post import constant folding and then transform this back into a GT_CALL during rationalization on older hardware.

However, given it is a more complex change I opted to push it out to a later PR.

Reply:
For Arm64, we could do the same or we could add basic SIMD constant folding support for `PopCount`, `AddAcross`, and `ToScalar`. `CreateScalarNode` will already generate a `GT_CNS_VEC` where applicable, including post import.